static/js/components/NewGroupUserForm.jsx

```diff
@@ -1,52 +1,105 @@
 import PropTypes from 'prop-types';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
+import TextField from '@material-ui/core/TextField';
+import Autocomplete, { createFilterOptions } from '@material-ui/lab/Autocomplete';
 
-import * as Action from '../ducks/groups';
+import * as groupsActions from '../ducks/groups';
+import * as usersActions from '../ducks/users';
+
+
+const filter = createFilterOptions();
 
 const NewGroupUserForm = ({ group_id }) => {
   const dispatch = useDispatch();
-  const [formState, setState] = useState({
-    newUserEmail: "",
+  const { allUsers } = useSelector((state) => state.users);
+  const [formState, setFormState] = useState({
+    newUserEmail: null,
     admin: false
   });
 
-  const handleChange = (event) => {
-    setState({
-      ...formState,
-      newUserEmail: event.target.value
-    });
-  };
+  useEffect(() => {
+    if (allUsers.length === 0) {
+      dispatch(usersActions.fetchUsers());
+    }
+  }, [dispatch, allUsers]);
 
   const handleSubmit = (event) => {
     event.preventDefault();
-    dispatch(Action.addGroupUser({
+    dispatch(groupsActions.addGroupUser({
       username: formState.newUserEmail,
       admin: formState.admin,
       group_id
     }));
-    setState({
-      newUserEmail: "",
+    setFormState({
+      newUserEmail: null,
       admin: false
     });
   };
 
   const toggleAdmin = (event) => {
-    setState({
+    setFormState({
       ...formState,
       admin: event.target.checked
     });
   };
 
   return (
     <div>
-      <input
-        type="text"
+      <Autocomplete
         id="newUserEmail"
         value={formState.newUserEmail}
-        onChange={handleChange}
+        onChange={(event, newValue) => {
+          if (typeof newValue === 'string') {
+            setFormState({
+              newUserEmail: newValue,
+            });
+          } else if (newValue && newValue.inputValue) {
+            // Create a new value from the user input
+            setFormState({
+              newUserEmail: newValue.inputValue,
+            });
+          } else {
+            setFormState({ newUserEmail: newValue.username });
+          }
+        }}
+        filterOptions={(options, params) => {
+          const filtered = filter(options, params);
+
+          // Suggest the creation of a new value
+          if (params.inputValue !== '') {
+            filtered.push({
+              inputValue: params.inputValue,
+              username: `Add "${params.inputValue}"`,
+            });
+          }
+
+          return filtered;
+        }}
+        selectOnFocus
+        clearOnBlur
+        handleHomeEndKeys
+        options={allUsers}
+        getOptionLabel={(option) => {
+          // Value selected with enter, right from the input
+          if (typeof option === 'string') {
+            return option;
+          }
+          // Add "xxx" option created dynamically
```

Comment:
What does this do?

Reply:
This adds the visible option that says "Add 'xxx'" where 'xxx' is whatever the user typed in (see screen capture gif to see this in action). This is the standard recipe provided by mui

Reply:
Ah, ok, I didn't know what the XXX meant---maybe just clarify that.